File: algorithms/strings/is_unique/IsUnique.java
import java.util.Arrays;
import java.util.List;

public class IsUnique {

    private static boolean isUnique(String s) {
        if (s.length() > 128)
            return false;

        boolean[] charSet = new boolean[128];

        for (char ch : s.toCharArray()) {
            if (charSet[ch] == true)
                return false;

            charSet[ch] = true;
        }

        return true;
    }

    public static void main(String[] args) {
        List<String> words = Arrays.asList("abcde", "hello", "apple", "kite", "padle");

        for (String word : words)
            System.out.printf("%s : %s\n", word, isUnique(word));
    }

}

File: algorithms/strings/is_unique/is_unique.rs
fn is_unique(s: &str) -> bool {
    if s.len() > 128 {
        return false
    }

    let mut char_set: [bool;128] = [false;128];

    for ch in s.chars() {
        if char_set[ch as usize] {
            return false;
        }
        
        char_set[ch as usize] = true;
    }

    true
}


fn main() {

    let words: [String;5] = [
        String::from("abcde"),
        String::from("hello"),
        String::from("apple"), 
        String::from("kite"), 
        String::from("padle")    
    ];

    for word in words.iter() {
        println!("{} : {}", word,is_unique(&word));
    }

}